src/cabinetry/model_utils.py

```diff
@@ -318,68 +319,79 @@ def yield_stdev(
         # add total prediction (sum over samples)
         down_comb = np.vstack((down_comb, np.sum(down_comb, axis=0)))
         # turn into list of channels
-        down_yields = [
+        down_yields_per_channel = [
             down_comb[:, model.config.channel_slices[ch]]
             for ch in model.config.channels
         ]
-        # append list of yields summed per channel
-        down_yields += [
-            np.asarray(np.sum(chan_yields, axis=-1, keepdims=True))
-            for chan_yields in down_yields
-        ]
+
+        # calculate list of yields summed per channel
+        down_yields_sum = np.stack(
+            [
+                np.sum(chan_yields, axis=-1, keepdims=True)
+                for chan_yields in down_yields_per_channel
+            ]
+        )
+        down_yields_sum = down_yields_sum.reshape(down_yields_sum.shape[:-1]).T
+
+        down_yields = np.concatenate((down_comb, down_yields_sum), axis=1)
         down_variations.append(down_yields)
 
-    # convert to awkward arrays for further processing
-    up_variations_ak = ak.from_iter(up_variations)
-    down_variations_ak = ak.from_iter(down_variations)
+    # convert to numpy arrays for further processing
+    up_variations_np = np.array(up_variations)
+    down_variations_np = np.array(down_variations)
```

Comment:
```suggestion
    up_variations_np = np.asarray(up_variations)
    down_variations_np = np.asarray(down_variations)
```

Usually `asarray` vs `array` will try to avoid copies if not needed. In this case the copy is actually needed regardless, so they end up doing the same. I find `asarray` still a bit more clear to indicate that we do not rely on the copy here (and it is more consistent with the rest of the codebase).